frame/treasury/src/lib.rs

```diff
@@ -210,10 +260,12 @@ decl_module! {
 			Approvals::mutate(|v| v.push(proposal_id));
 		}
 
+		/// This function will implement the `OnFinalize` trait
 		fn on_finalize(n: T::BlockNumber) {
 			// Check to see if we should spend some funds!
 			if (n % T::SpendPeriod::get()).is_zero() {
-				Self::spend_funds();
+				Self::spend_funds(Funds::Ring);
+				Self::spend_funds(Funds::Kton);
```

Comment:
The Funds enums seem to only be used here? If so, let's keep it simple by just using 

```
Self::spend_ring_funds();
Self::spend_kton_funds();
```

?